### PR TITLE
ntp-units.d support

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by mbiebl** [comment]
I want to add here that b72ddf0 mentions that alternative NTP implementations should use `Conflicts=systemd-timesyncd.service` to take over responsibility for the time sync.
Unfortunately, this mechanism does not work reliably.
See also
https://github.com/systemd/systemd/issues/7104#issuecomment-453783756
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=902026

**Comment by keszybz** [comment]
Actually, I had it on my TODO lists to add Conflicts to `systemd-timesync.service`. It works as a backup mechism to make sure that more than one ntp service is not started at the same time. But Conflicts are symmetrical, so they don't solve the issue of precedence or selecting which unit to start.

File: man/systemd-timedated.service.xml [hunk 1]
```diff
@@ -46,25 +46,36 @@
...
+    <para><command>systemd-timesyncd</command> will look for files with a <literal>.list</literal> extension
+    in <filename>ntp-units.d/</filename> directories. Each file is parsed as a list of unit names, one per
+    line. Empty lines and lines with comments (<literal>#</literal>) are ignored.  Files are read from
+    <filename>/usr/lib/systemd/ntp-units.d/</filename> and the corresponding directories under
+    <filename>/etc/</filename>, <filename>/run/</filename>, <filename>/usr/local/lib/</filename>. Files in
+    <filename>/etc/</filename> override files with the same name in <filename>/run/</filename>,
+    <filename>/usr/local/lib/</filename>, and <filename>/usr/lib/</filename>. Files in
+    <filename>/run/</filename> override files with the same name under <filename>/usr/</filename>. Packages
+    should install their configuration files in <filename>/usr/lib/</filename> (distribution packages) or
+    <filename>/usr/local/lib/</filename> (local installs).</para>
```
Comment (boucman) [inline man/systemd-timedated.service.xml:60]: reading this was a bit confusing... you explain how to build the list of timesync services but you don't really say what they are used for anywhere. I assume that they become some sort of dependency for systemd-timesyncd but that's never explicitely stated

is there a generator that reads them at early boot, or are they added to the transaction ?

**Comment by poettering** [comment]
urks, please no. Let's not add cruft back we already removed once because it's crufty. If ntp implementations don't use the env var, then maybe fix that, but let's not add this awful hack again. Please!

**Comment by keszybz** [comment]
I don't think it should be called crufty. The implementation is ~50 lines of code, the format and its description are dead simple. They are also dead simple to use correctly for distros, and in fact as soon as the support is added in systemd, things "just work" in Fedora. As I wrote in the commit message (and as was discussed in the previous ticket), the env vars are complicated to use correctly. We are building a (tool to build a) distro, not a standalone thing, so the effect on the whole ecosystem should be considered: if we remove a bit of clean code from systemd, but add equally complex code in installation scriptlets, then we lose.

**Comment by poettering** [comment]
still think this is awful as hell... but ok...

(a way to make this more digestable might be to split out this logic into a generic call, I'd bet we have a logic of "read all lines in all files in these dirs and remove comments" somewhere else...)

**Pull request merged**